phdi/fhir/conversion.py

```
@@ -0,0 +1,184 @@
+    """
+    Private helper function to determine what settings to use with the
+    FHIR server to facilitate message conversion. Some data streams
+    will be encoded in HL7 format, whereas others will be XML data.
+    Attempts to identify which data type the input has and determine
+    the appropriate FHIR converter root template to use. If the user
+    opts to not use the default CCDA root template in cases where an
+    input resource isn't supported, the functionwill raise an
+    exception if a message's extracted LOINC code doesn't correspond to
+    an existing CCDA template. More information about the required
+    templates and settings can be found here:
+
+    https://docs.microsoft.com/en-us/azure/healthcare-apis/azure-api-for-fhir/convert-data
+
+    :param message: The incoming message (already cleaned, if
+      applicable)
+    :param use_default_ccda: Optionally, whether to default to the
+      base "CCD" root template if a resource's LOINC code doesn't
+      map to a specific supported template. Default is No.
+    :return: A dictionary holding the settings of parameters to-be
+      set when converting the input to FHIR
+    """
+    # Some streams (e.g. ELR, VXU) are HL7v2 encoded
+    if message[:3] == "MSH":
+        try:
+            parsed_msg = hl7.parse(message)
+            extracted_code = str(parsed_msg.segment("MSH")[9])
+
+            # HL7 MSH segment 9 has three components: message code, trigger
+            # event, and message structure. We can extract based on number of
+            # present separators and recombine to create a robust formatted code
+            extracted_code_tokenized = extracted_code.split(parsed_msg.separators[3])
+            formatted_code = ""
+            if (len(extracted_code_tokenized) >= 3) and (
+                extracted_code_tokenized[2] != ""
+            ):
+                formatted_code = extracted_code_tokenized[2]
+            elif len(extracted_code_tokenized) == 2:
```

Nit/question: should we raise an exception if we can't derive message structure?